src/components/ebay-snackbar-dialog/index.marko

```diff
@@ -0,0 +1,105 @@
+import processHtmlAttributes from "../../common/html-attributes"
+
+static var ignoredAttributes = [
+    "open"
+];
+
+class {
+    clearAllTimeouts() {
+        while(this.timeouts.length) {
+            clearTimeout(this.timeouts.pop());
+        }
+    }
+
+    onInput(input) {
+        this.clearAllTimeouts();
+        this.state = {open: input.open || this.state.open || false}
+        if (input.open !== false) {
+            this.timeouts.push(setTimeout(() => {
+                this.emit("close");
+                this.state.open = false;
+            }, input.timeout || this.DEFAULT_TIMEOUT_LENGTH));
+        }
+    }
+    onDestroy() {
+        this.clearAllTimeouts();
+    }
+
+    onCreate() {
+        this.DEFAULT_TIMEOUT_LENGTH = 6000;
+        this.eventSet = new Set();
+        this.state = {open: false};
+        this.timeouts = [];
+    }
+
+    handleAction() {
+        this.clearAllTimeouts();
+        this.emit('action');
+        this.state.open = false;
+        this.emit('close');
+    }
+
+    handleFocus() {
+        this.clearAllTimeouts();
+        this.eventSet.add("focus");
+    }
+
+    handleBlur() {
+        this.eventSet.delete("focus");
+        if (this.eventSet.size === 0) {
+            this.timeouts.push(setTimeout(() => {
+                this.emit("close");
+                this.state.open = false;
+            }, this.DEFAULT_TIMEOUT_LENGTH));
+        }
+    }
+
+    handleMouseEnter() {
+        this.clearAllTimeouts();
+        this.eventSet.add("mouseOver");
+    }
+
+    handleMouseLeave() {
+        this.eventSet.delete("mouseOver");
```

Comment:
Should we not clear all timeoutes here and `handleBlur` as well?
This is related to the comment about the timeout array.

IMO this is what should happen.
* Anytime we add a new timeout, we clear the old one.
* Anytime we want to delay the timeout, we clear the old one
* On destroy clears the old one
This way we probably don't need multiple timeouts.